libevm/EVMC.cpp

```diff
@@ -44,19 +44,47 @@ owning_bytes_ref EVMC::exec(u256& io_gas, ExtVMFace& _ext, const OnOpFunc& _onOp
         return VMFactory::create(VMKind::Legacy)->exec(io_gas, _ext, _onOp);
     }
 
-    // TODO: Add EVM-C result codes mapping with exception types.
-    if (r.status() == EVM_FAILURE)
+    const evm_status_code status = r.status();
+    assert(status >= 0);  // Do not allow internal errors to pass through.
+
+    if (status == EVM_SUCCESS || status == EVM_REVERT)
```

Comment:
Why not move the above `EVM_REJECTED` case here too? `if rejected else if success|revert else`

Reply:
Or perhaps move the entire thing into the switch statement.

Reply:
Let's see...